03-task-manager/starter/controllers/tasks.js

```diff
@@ -0,0 +1,47 @@
+const Task = require('../models/Task')
+var ObjectId = require('mongoose').Types.ObjectId;
+
+
+const allTasks = async (req, res) => {
+    const tasks = await Task.find({})
+    res.status(200).json({ tasks })
+}
+
+const newTask = async (req, res) => {
+    const task = await Task.create(req.body)
+    res.status(201).json({ "Message": 'Succeeded', "data": task })
+}
+
+const getTask = async (req, res) => {
+    const id = new ObjectId(req.params.id)
+    const task = await Task.find({ _id: id }).exec()
```

Comment:
The way this is implemented right now, you will never get to the not found error. The find method will return an empty array if the task is not found and that doesn't evaluate to false. Instead, you can use the `findById` method, which returns a single task and will return a falsy value if the task is not found.

```suggestion
    const task = await Task.findById(id).exec();
```